flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/dataview/ListView.java

```diff
@@ -186,22 +182,4 @@ public static DataType newListViewDataType(DataType elementDataType) {
                 ListView.class,
                 DataTypes.FIELD("list", DataTypes.ARRAY(elementDataType).bridgedTo(List.class)));
     }
-
-    // --------------------------------------------------------------------------------------------
-    // Legacy
-    // --------------------------------------------------------------------------------------------
-
-    @Deprecated public transient TypeInformation<?> elementType;
-
-    /**
-     * Creates a {@link ListView} for elements of the specified type.
-     *
-     * @param elementType The type of the list view elements.
-     * @deprecated This method uses the old type system. Please use a {@link DataTypeHint} instead
-     *     if the reflective type extraction is not successful.
```

Comment:
Should this hint be kept in javadoc somewhere? Feels like useful advice for people migrating code

Reply:
Thanks @SamBarker . I added back the note.